src/core/init/manifest_update_scheduler.ts

```diff
@@ -20,103 +20,159 @@ import {
+
+    const timeSinceRequest = sendingTime == null ? 0 :
+                                                   performance.now() - sendingTime;
+    const minInterval = Math.max(minimumManifestUpdateInterval - timeSinceRequest, 0);
+
+    const autoRefresh$ = manifest.lifetime === undefined || manifest.lifetime <= 0 ?
+      EMPTY :
+      observableTimer(Math.max(manifest.lifetime * 1000 - timeSinceRequest,
+                               minInterval)).pipe(mapTo(false));
+
+    const expired$ = manifest.expired === null ?
+      EMPTY :
+      observableTimer(minimumManifestUpdateInterval)
+        .pipe(mergeMapTo(observableFrom(manifest.expired)),
+              mapTo(true));
+
+   const maxInterval$ = maximumManifestUpdateInterval === undefined ?
+     EMPTY :
+     observableTimer(maximumManifestUpdateInterval).pipe(mapTo(false));
 
     // Emit when the manifest should be refreshed. Either when:
     //   - A buffer asks for it to be refreshed
     //   - its lifetime expired.
-    return observableMerge(autoRefresh$, manualRefresh$, expired$)
-      .pipe(take(1),
-            mergeMap(() => refreshManifest(initialManifest.manifest,
-                                           fetchManifest)),
-            mergeMap(handleManifestRefresh$),
-            ignoreElements());
+    return observableMerge(autoRefresh$, manualRefresh$, expired$, maxInterval$).pipe(
+      take(1),
+      mergeMap(refreshManifest),
+      mergeMap(handleManifestRefresh$),
+      ignoreElements());
   }
 
   return observableDefer(() => handleManifestRefresh$(initialManifest));
+
+
+  /**
+   * Refresh the Manifest.
+   * Perform a full update if a partial update failed.
+   * @param {boolean} completeRefresh
+   * @returns {Observable}
+   */
+   function refreshManifest(
+     completeRefresh : boolean
+   ) : Observable<IFetchManifestResult> {
+     const fullRefresh = completeRefresh || manifestUpdateUrl === undefined;
+     const refreshURL = fullRefresh ? manifest.getUrl() :
+                                      manifestUpdateUrl;
+     if (!isNonEmptyString(refreshURL)) {
+       log.warn("Init: Cannot refresh the manifest: no url");
+       return EMPTY;
+     }
+     const externalClockOffset = manifest.getClockOffset();
+     return fetchManifest(refreshURL, externalClockOffset)
+       .pipe(mergeMap(({ manifest: newManifest, sendingTime: newSendingTime }) => {
+         if (completeRefresh) {
+           manifest.replace(newManifest);
+           return observableOf({ manifest, sendingTime: newSendingTime });
+         }
+         try {
+           manifest.update(newManifest);
+         } catch (e) {
+           const message = e instanceof Error ? e.message :
+                                                "unknown error";
+           log.warn(`MUS: Attempt to update Manifest failed: ${message}`,
+                    "Re-downloading the Manifest fully");
+           return startManualRefreshInterval(true,
```

Comment:
So here, you consider new sending time to re-schedule manifest ? Why ?

Reply:
Yep. I would say that the `OUT_OF_SYNC_MANIFEST_REFRESH_DELAY` value is mostly here to avoid doing too much Manifest requests (here for perf reasons). It makes sense to wait from the last failed update.